Find a functionally equivalent code:android.util.LongSparseLongArray.put:COMMENT
Method Modifier: public      
Comment:/**
 * Adds a mapping from the specified key to the specified value,
 * replacing the previous mapping from the specified key if there
 * was one.
 */

@@ -4,18 +4,11 @@
         mValues[i] = value;
     } else {
         i = ~i;
-        if (mSize >= mKeys.length) {
-            growKeyAndValueArrays(mSize + 1);
-        }
-        if (mSize - i != 0) {
-            System.arraycopy(mKeys, i, mKeys, i + 1, mSize - i);
-            System.arraycopy(mValues, i, mValues, i + 1, mSize - i);
-        }
-        mKeys[i] = key;
-        mValues[i] = value;
+        mKeys = GrowingArrayUtils.insert(mKeys, mSize, i, key);
+        mValues = GrowingArrayUtils.insert(mValues, mSize, i, value);
         mSize++;
     }
 }

